kernel/CMakeLists.txt

```diff
@@ -28,7 +28,10 @@ target_sources_ifdef(CONFIG_INT_LATENCY_BENCHMARK kernel PRIVATE int_latency_ben
 target_sources_ifdef(CONFIG_STACK_CANARIES        kernel PRIVATE compiler_stack_protect.c)
 target_sources_ifdef(CONFIG_SYS_CLOCK_EXISTS      kernel PRIVATE timer.c)
 target_sources_ifdef(CONFIG_ATOMIC_OPERATIONS_C   kernel PRIVATE atomic_c.c)
-target_sources_ifdef(CONFIG_PTHREAD_IPC           kernel PRIVATE pthread.c)
+target_sources_ifdef(CONFIG_PTHREAD_IPC           kernel PRIVATE posix/pthread_common.c)
+target_sources_ifdef(CONFIG_PTHREAD_IPC           kernel PRIVATE posix/pthread_cond.c)
+target_sources_ifdef(CONFIG_PTHREAD_IPC           kernel PRIVATE posix/pthread_mutex.c)
+target_sources_ifdef(CONFIG_PTHREAD_IPC           kernel PRIVATE posix/pthread_barrier.c)
```

Comment:
add CMakeLists.txt inside posix directory and do all of that in there using one target_sources() command